package/libpiksi/libpiksi/src/util.c

```diff
@@ -814,12 +825,10 @@ static pipeline_t *pipeline_wait(pipeline_t *r)
   char *output = r->stdout_buffer;
 
   while (output_size > total) {
-    ssize_t ret = read(out_fd, output + total, output_size - total);
-    if (ret > 0) {
-      total += ret;
-    } else {
-      break;
-    }
+    ssize_t read_result = read(out_fd, output + total, output_size - total);
+    if (read_result < 0) break;
```

Comment:
Note that this will stay in the loop infinitely if nothing is being read, while the old code broke out in case `ret == 0`. Not sure if it's an issue

Reply:
Good catch :-), will fix